### PR TITLE
Fix tslint version 5

### PR DESCRIPTION
**Pull request by ebenoist**
I think this applies most of the changes originally proposed by @KolesnichenkoDS with the changes @blueyed suggested in https://github.com/neomake/neomake/pull/1204

I have tested the change locally with tslint 5.3

### CODE REVIEW

**Comment by blueyed** [comment]
There is a failing test:

```
    (319/338) [EXECUTE] typescript: tsc
    (319/338) [EXECUTE] (X) Vim(lgetexpr):E377: Invalid %E in format string
      > /home/travis/build/neomake/neomake/tests/ft_typescript.vader:9: lgetexpr "t.ts(1,7): error TS2322: Type '\"hello\"' is not assignable to type 'Number'."
```

You should be able to run them using `make tests/ft_typescript.vader`.

**Comment by blueyed** [comment]
There's a missing comma:

```diff
diff --git i/autoload/neomake/makers/ft/typescript.vim w/autoload/neomake/makers/ft/typescript.vim
index 7aa92ebe..3b5698b4 100644
--- i/autoload/neomake/makers/ft/typescript.vim
+++ w/autoload/neomake/makers/ft/typescript.vim
@@ -10,7 +10,7 @@ function! neomake#makers#ft#typescript#tsc() abort
         \ 'args': ['--project', neomake#utils#FindGlobFile('tsconfig.json'), '--noEmit', '--watch', 'f
         \ 'append_file': 0,
         \ 'errorformat':
-            \ '%E%f(%l\,%c): %m' .
+            \ '%E%f(%l\,%c): %m,' .
             \ '%E%f %#(%l\,%c): error %m,' .
             \ '%E%f %#(%l\,%c): %m,' .
             \ '%Eerror %m,' .
```

But then the test fails, since "error" is included in the message then.

For this the following would help:
```
diff --git i/autoload/neomake/makers/ft/typescript.vim w/autoload/neomake/makers/ft/typescript.vim
index 7aa92ebe..e3194043 100644
--- i/autoload/neomake/makers/ft/typescript.vim
+++ w/autoload/neomake/makers/ft/typescript.vim
@@ -10,7 +10,7 @@ function! neomake#makers#ft#typescript#tsc() abort
         \ 'args': ['--project', neomake#utils#FindGlobFile('tsconfig.json'), '--noEmit', '--watch', 'f
         \ 'append_file': 0,
         \ 'errorformat':
-            \ '%E%f(%l\,%c): %m' .
+            \ '%E%f(%l\,%c): error %m,' .
             \ '%E%f %#(%l\,%c): error %m,' .
             \ '%E%f %#(%l\,%c): %m,' .
             \ '%Eerror %m,' .
```

But I think there should be some more tests covering the new format(s).

**Comment by ebenoist** [comment]
@blueyed added your comma fix. I can take a look at specs later this week.

**Comment by blueyed** [comment]
Thanks.
Some example output, especially for the new errorformat would be all that is needed to add the tests.

**Comment by ebenoist** [comment]
@blueyed updated with new tests for both tslint and tsc

File: autoload/neomake/makers/ft/typescript.vim [hunk 1]
```diff
@@ -10,6 +10,7 @@ function! neomake#makers#ft#typescript#tsc() abort
         \ 'args': ['--project', neomake#utils#FindGlobFile('tsconfig.json'), '--noEmit', '--watch', 'false'],
         \ 'append_file': 0,
         \ 'errorformat':
+            \ '%E%f(%l\,%c): error %m,' .
             \ '%E%f %#(%l\,%c): error %m,' .
```
Comment (blueyed) [inline autoload/neomake/makers/ft/typescript.vim]: The new errorformat is the same as the previous one below it basically.
` %#` translates to "at least one space".
Why is the new one needed?

Comment (blueyed) [inline autoload/neomake/makers/ft/typescript.vim]: See https://github.com/neomake/neomake/pull/1204#discussion_r111663030.

Comment (ebenoist) [inline autoload/neomake/makers/ft/typescript.vim]: I don't know -- I copied this from #1204. I never experienced issues with the tsc checker, just the tslint checker.

Comment (blueyed) [inline autoload/neomake/makers/ft/typescript.vim]: Ok, I see then.
Will remove it when merging it.

Comment (ebenoist) [inline autoload/neomake/makers/ft/typescript.vim]: 👍

**Comment by blueyed** [comment]
Thanks for taking this!

**Comment by ebenoist** [comment]
No worries, many thanks for merging!

**Pull request closed without merging**